react/.storybook/main.js

```diff
@@ -8,29 +9,55 @@ module.exports = {
     '@storybook/addon-viewport',
     '@storybook/addon-storysource',
     '@storybook/addon-a11y',
+    {
+      name: '@storybook/preset-create-react-app',
+    },
   ],
   webpackFinal: async (config, { configType }) => {
+    
+    // Remove the default sass-loader rule from create-react-app.
+    config.module.rules = config.module.rules.filter(rule => {
+      if (rule.test) {
+        if (!'some.scss'.match(rule.test)) {
+          return true;
+        } else {
+          return rule.oneOf;
+        }
+      } else {
+        return true;
+      }
+    });
+    // @storybook/preset-create-react-app changes the entire
+    // structure of rules. one of the rules with oneOf
+    // contains all the rules now.
+    const groupedRules = config.module.rules.find((rule) => {
+      if (rule.oneOf) {
+        return rule;
+      }
+    });
     // Add special plugins to the babel loader.
-    const babelRule = config.module.rules.find(rule => rule.loader && rule.loader.match('babel-loader'));
-    babelRule.options.plugins = babelRule.options.plugins.concat([
-      "@babel/proposal-export-default-from",
-      '@babel/proposal-class-properties',
-      '@babel/plugin-proposal-object-rest-spread'
-    ]);
-
-    // Alter the mini-css-extract-plugin to look for asset files relative to
-    // the storybook-static directory.
-    if ( configType === 'PRODUCTION') {
-      const cssRule = config.module.rules.find(rule => rule.test && 'some.css'.match(rule.test));
-      if(cssRule) {
-        const cssFileLoader = cssRule.use.find(loader => loader.loader && loader.loader.match('mini-css-extract-plugin'));
-        if(cssFileLoader) {
-          cssFileLoader.options.publicPath = (resourcePath, context) => {
-            // publicPath is the relative path of the resource to the context
-            // e.g. for ./css/admin/main.css the publicPath will be ../../
-            // while for ./css/main.css the publicPath will be ../
-            return path.relative(path.dirname(resourcePath), context) + '/';
-          };
+    const babelRule = groupedRules.oneOf.find(rule => rule.loader && rule.loader.match('babel-loader'));
+    if (babelRule) {
+      babelRule.options.plugins = babelRule.options.plugins.concat([
+        "@babel/proposal-export-default-from",
+      ]);
+      // Remove preset-env as preset-react-app already defines it.
+      babelRule.options.presets = babelRule.options.presets.filter(plugin => !plugin.includes('/preset-env/'));
```

Comment:
are these two babel plugins come with create react app? what do they do? 
also, can you combine these two into one? 

babelRule.options.presets = babelRule.options.presets.filter(plugin => !plugin.includes('/preset-env/') && !plugin.includes('/preset-react/'));

Reply:
Do we still need babel config in nwb? 
```
  babel: {
    stage: 1,
    plugins: [
      ['module-resolver', {
        root: ['./src'],
        alias: {
          SharedAssets: '@massds/mayflower-assets/static'
        }
      }]
    ]
  }
```

Reply:
[babel/proposal-export-default-from ](https://babeljs.io/docs/en/babel-plugin-proposal-export-default-from)does not come with [babel/preset-create-react-app](https://www.npmjs.com/package/babel-preset-react-app) and is needed to support the `export ... from` syntax:
```
export Form, { FormProvider } from './components/forms/Form';
```
[babel/preset-env](https://babeljs.io/docs/en/babel-preset-env) provides polyfills for cross browser support.
[babel/preset-react](https://babeljs.io/docs/en/babel-preset-react) provides support for react transpilation.

Reply:
Nope, we can remove the babel settings in nwb.config.